app/services/github/client.rb

```diff
@@ -0,0 +1,98 @@
+module Github
+  # Github client (uses ocktokit.rb as a backend)
+  class Client
+    class << self
+      # adapted from https://api.rubyonrails.org/classes/Module.html#method-i-delegate_missing_to
+      def method_missing(method, *args, &block)
+        return super unless target.respond_to?(method, false)
+
+        # define for re-use
+        self.class.define_method(method) do |*new_args, &new_block|
+          request do
+            target.public_send(method, *new_args, &new_block)
+          end
+        end
+
+        # call the original method, this will only be called the first time
+        # as in subsequent calls, the newly defined method will prevail
+        request do
+          target.public_send(method, *args, &block)
+        end
+      end
+
+      # adapted from https://api.rubyonrails.org/classes/Module.html#method-i-delegate_missing_to
+      def respond_to_missing?(method, _include_all = false)
+        target.respond_to?(method, false) || super
+      end
+
+      private
+
+      def request
+        Honeycomb.add_field("app.name", "github.client")
```

Comment:
@mstruve let me know if these are fine, same in `record_error`. I mirrored the same structure of `Search::Client`

Reply:
This is actually not going to work how you want it to. For example, with searching, I thought it would label the search span itself but that span is not created until after our HTTP client starts making the request much further down the stack so the only thing we end up adding a field to is the controller method or sidekiq worker span that is implementing the request. Its not hurting anything so I left it but we need to dig deep into how HTTP requests are tracked if we want to add to their spans.